### DISCUSSION TITLE
[Support] 新建的midwayjs项目为什么使用jest会报错？

### DISCUSSION THREAD

**Discussion by MakerLi1994** [midwayjs/midway#3687] [Q&A]
### Describe the problem（描述问题）

# 操作步骤

1. 下载最新包

`npm init midway@latest -y`

2. 选择  egg-v3
3. 进入项目并下载依赖
`cd midway-project
cnpm install
`
4. package.json新增jest执行命令
`"nodetest": "jest",`
5. 运行单测
` npm run  nodetest`
6. 抛出异常
```triple quotes blocks:
  console.error
      TypeError: Cannot use 'in' operator to search for 'enable' in 105
          at EggAgentWorkerLoader.normalizePluginConfig (D:\eggtest\midway-project\node_modules\_egg-core@4.31.0@egg-core\lib\loader\mixin\plugin.js:219:20)
          at EggAgentWorkerLoader.readPluginConfigs (D:\eggtest\midway-project\node_modules\_egg-core@4.31.0@egg-core\lib\loader\mixin\plugin.js:194:14)
```
7. 删除对应文件
src>config>plugin.ts
8.再次执行单测
9.运行正常
` PASS  test/controller/api.test.ts
 PASS  test/controller/home.test.ts
A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --detectOpenHandles to find leaks. Active timers can also cause this, ensure that .unref() was called on them.

Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        5.208 s`

### Midway Versions（Midway 版本）

{
 "name": "my-midway-project", 
 "version": "1.0.0", 
 "description": "", 
 "private": true, 
 "dependencies": {
  "@midwayjs/bootstrap": "^3.12.0", 
  "@midwayjs/core": "^3.12.0", 
  "@midwayjs/web": "^3.12.0", 
  "@midwayjs/decorator": "^3.12.0", 
  "@midwayjs/logger": "^3.1.0", 
  "egg": "^2.0.0"
 }, 
 "devDependencies": {
  "@midwayjs/egg-ts-helper": "^1.0.1", 
  "@midwayjs/mock": "^3.12.0", 
  "@types/jest": "^29.2.0", 
  "@types/node": "14", 
  "cross-env": "^6.0.0", 
  "egg-mock": "^3.26.0", 
  "mwts": "^1.3.0", 
  "mwtsc": "^1.4.0", 
  "jest": "^29.2.2", 
  "ts-jest": "^29.0.3", 
  "ts-node": "^10.9.2", 
  "typescript": "~4.8.0"
 }, 
 "engines": {
  "node": ">=12.0.0"
 }, 
 "scripts": {
  "nodetest": "jest"
 }
}

**Comment by czy88840616** [comment]
这是 egg 的报错，你必须使用 ts-node 来执行。

```
"test": "cross-env NODE_ENV=unittest NODE_OPTIONS=--require=ts-node/register jest",
```
